src/Compilers/CSharp/Portable/Binder/ForEachLoopBinder.cs

```diff
@@ -992,36 +997,29 @@ private void GetDisposalInfoForEnumerator(ref ForEachEnumeratorInfo.Builder buil
             // If IDisposable is available but its Dispose method is not, then diagnostics will be reported only if the enumerator
             // is potentially disposable.
 
+            // For async foreach, we don't do the runtime check
```

Comment:
>// For async foreach, we don't do the runtime check [](http://example.com/codeflow?start=12&length=51)

Consider moving to line 1045, after the following, so it's clear what this refers to.
```
// if it wasn't pattern-disposable, see if it's directly convertable to IDisposable
```